numpy/lib/tests/test_arraysetops.py

```diff
@@ -289,6 +289,75 @@ def test_in1d(self):
 
         assert_array_equal(in1d([], []), [])
 
+    def test_in1d_hit_alternate_algorithm_floats(self):
+        # Perform the above test but with floats
+        # This forces the old in1d (pre-Oct 1/2018) algorithm
```

Comment:
It would be nice to have some clearer way to force one algorithm or the other, just so that we can use the same contrived corner-cases for each in tests.

Reply:
Should I add some debug argument to isin/in1d for this? I agree with your comment.

Reply:
Added a flag to isin/in1d for this and extended the tests.

Reply:
It would be nice if you could make it a 3-way switch. `None` auto detects, `True`/`False` force one or the other.